src/libcore/ptr/mod.rs

```diff
@@ -1307,14 +1307,14 @@ macro_rules! fnptr_impls_safety_abi {
         #[stable(feature = "fnptr_impls", since = "1.4.0")]
         impl<Ret, $($Arg),*> fmt::Pointer for $FnTy {
             fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                fmt::Pointer::fmt(&(*self as *const ()), f)
+                fmt::Pointer::fmt(&(*self as usize as *const ()), f)
```

Comment:
Could you add a comment that explains why the intermediate cast is necessary? Preferably with a link to the issue for function pointer to type pointer casts on architectures with multiple address spaces.

Reply:
I have amended the commit that adds this hack with a comment and a link describing the issue.